Florida_Analysis_Py_File.py

```diff
@@ -0,0 +1,534 @@
+
+import matplotlib.pyplot as plt
+
+plt.xlim(-3, 3)
+# plt.ylim(0, 500)
+
+plt.xlabel("Year Relative To Policy Change Year (2010)")
+plt.ylabel("Deaths due to Drug Overdose (Per 100K)")
+plt.title("Pre-Post Analysis Of Drug Overdose Deaths in Florida")
+
+plt.plot(x_before, y_pred_before, color="blue")
+plt.plot(x_after, y_pred_after, color="red")
+
+plt.grid()
+# plot avg value in each year
+
+## Difference-In-Difference Analysis Of Drug Overdose Deaths In Florida
+
+# creating a new dataset, that contains drug related deaths in other neighboring states of Florida - Alabama, Georgia, Mississippi, South Carolin and Tennessee
+
+other_states_deaths = deaths_copy[
+    (deaths_copy["State"] != "Florida")
+    & (
+        (
+            deaths_copy["Drug/Alcohol Induced Cause"]
+            == "Drug poisonings (overdose) Unintentional (X40-X44)"
+        )
+        | (
+            deaths_copy["Drug/Alcohol Induced Cause"]
+            == "Drug poisonings (overdose) Undetermined (Y10-Y14)"
+        )
+        | (
+            deaths_copy["Drug/Alcohol Induced Cause"]
+            == "Drug poisonings (overdose) Suicide (X60-X64)"
+        )
+    )
+]
+other_states_deaths_copy = other_states_deaths.copy()
+
+other_states_deaths_copy["Overdose_Per_100K"] = (
+    other_states_deaths_copy["Deaths"] / other_states_deaths_copy["Population"]
+) * 100_000
+# calculating and displaying results for drug related deaths - year, state and county wise
+
+other_states_result = (
+    other_states_deaths_copy.groupby(["Year", "State", "County"])["Overdose_Per_100K"]
+    .sum()
+    .reset_index()
+)
+other_states_result.head()
+# calculating yearwise drug related deaths results for Alabama
+alabama_results_yearwise = (
+    other_states_deaths_copy[other_states_deaths_copy["State"] == "Alabama"]
+    .groupby("Year")["Overdose_Per_100K"]
+    .mean()
+)
+alabama_results_yearwise
+# calculating yearwise drug related deaths results for Georgia
+georgia_results_yearwise = (
+    other_states_deaths_copy[other_states_deaths_copy["State"] == "Georgia"]
+    .groupby("Year")["Overdose_Per_100K"]
+    .mean()
+)
+georgia_results_yearwise
+# calculating yearwise drug related deaths results for Mississippi
+mississippi_results_yearwise = (
+    other_states_deaths_copy[other_states_deaths_copy["State"] == "Mississippi"]
+    .groupby("Year")["Overdose_Per_100K"]
+    .mean()
+)
+mississippi_results_yearwise
+# calculating yearwise drug related deaths results for South Carolina
+south_carolina_results_yearwise = (
+    other_states_deaths_copy[other_states_deaths_copy["State"] == "South Carolina"]
+    .groupby("Year")["Overdose_Per_100K"]
+    .mean()
+)
+south_carolina_results_yearwise
+# calculating yearwise drug related deaths results for Tennessee
+tennessee_results_yearwise = (
+    other_states_deaths_copy[other_states_deaths_copy["State"] == "Tennessee"]
+    .groupby("Year")["Overdose_Per_100K"]
+    .mean()
+)
+tennessee_results_yearwise
```

Comment:
Hey Annapurna, it looks like you haven't changed the comp states for Florida. We can better comp states selection according to the comments given in the feedback of our first draft.

Reply:
Thanks Zhonglin! I will work on resolving this issue.

Reply:
No problem! Just letting me know if you have any questions!